server/src/main/java/org/opensearch/action/search/SearchResponse.java

```diff
@@ -112,6 +115,11 @@ public SearchResponse(StreamInput in) throws IOException {
         clusters = new Clusters(in);
         scrollId = in.readOptionalString();
         tookInMillis = in.readVLong();
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            phaseTook = new PhaseTook(in);
+        } else {
+            phaseTook = PhaseTook.NULL;
```

Comment:
We must define phaseTook in the constructor else it will be undefined. To match SearchResponse style, phaseTook is final and uninitialized.

```
    private final SearchResponseSections internalResponse;
    private final String scrollId;
    private final String pointInTimeId;
    private final int totalShards;
    private final int successfulShards;
    private final int skippedShards;
    private final ShardSearchFailure[] shardFailures;
    private final Clusters clusters;
    private final long tookInMillis;
    private final PhaseTook phaseTook;
```

Reply:
Yeah ignore my comments, I misread it. You are not anyway reading from stream, so its fine.